Tribler/Core/Modules/versioncheck_manager.py

```diff
@@ -44,5 +47,5 @@ def parse_response(response):
 
         agent = Agent(reactor)
         return agent.request('GET', VERSION_CHECK_URL, Headers({'User-Agent': ['Tribler ' + version_id]}), None)\
-            .addCallback(parse_response)\
-            .addCallback(parse_body)
+            .addCallbacks(parse_response, on_request_error)\
+            .addCallbacks(parse_body, on_request_error)
```

Comment:
two commentse here:

 - The errbacks are right next to the callbacks, so the first one (line 50) will be called when the request fails. the econd one (line 51) will trigger when parse_response fails. and if parse_body fails, nothing will catch the failure.

 - on_request_error is the equivalent of a catch all try/except. Not only that, but it does nothing to attempt to deal with the situation, so let's say the request fails. Instead of  `parse_response`, `on_request_error` will be called, which will log an entry and return `None`, as `None` is not a failure, it will be passed to `parse_body` which will crash and burn.

Have a look at how the `Deferred` callback chain works and trap specific (expected) failures only.